### PR TITLE
[KEYCLOAK-11409] Add GolangCI to Keycloak Operator repository

### PR DESCRIPTION
**Pull request by abstractj**
## Additional Information
Fixes issues on https://github.com/keycloak/keycloak-operator/pull/6
[GolangCI documentation](https://github.com/golangci/golangci-lint)
[Deprecation of gometalinter](https://github.com/alecthomas/gometalinter)

## Verification Steps

Here's the output: 
- https://github.com/abstractj/keycloak-operator/pull/1
- https://golangci.com/r/github.com/abstractj/keycloak-operator/pulls/1

## Additional Notes

I had to run `go mod tidy` to clean up unused dependencies and also noticed some timeouts while fetching from `git.apache.org`.

In summary the issue with GolangCI was the fact that checksums had to be updated. The issue was somewhat related to this: https://github.com/golangci/golangci/issues/25

We've been using GolangCI on Keycloak for some months with no issues, but if this starts to become a bottleneck we can revisit the setup or remove.

### CODE REVIEW

**Comment by pb82** [comment]
@abstractj I have a similar problem locally: `go: git.apache.org/thrift.git@v0.12.0: unknown revision v0.12.0`. This can be resolved by adding `replace git.apache.org/thrift.git => github.com/apache/thrift v0.12.0` to the end of the go.mod file as described here: https://github.com/operator-framework/operator-sdk/issues/1873

I didn't submit a PR for this because it looked like no one else had this problem.

**Comment by davidffrench** [comment]
Fantastic work @abstractj , this is an excellent check to have on the PRs. Small question, is it setup to also comment on the PRs with issues found, like I see on the keycloak-gatekeeper PRs e.g. https://github.com/keycloak/keycloak-gatekeeper/pull/473

Reading through the issue Peter linked, it looks like the git.apache.org infrastructure has had serious issues. I don't see any issues with adding `replace git.apache.org/thrift.git => github.com/apache/thrift v0.12.0` to the go.mod file now, it would be frustrating to have our CI fail randomly.

**Comment by slaskawi** [comment]
LGTM, @davidffrench @abstractj Since there's an ongoing discussion, I'm leaving merging this one up to you guys.

**Comment by abstractj** [comment]
> Fantastic work @abstractj , this is an excellent check to have on the PRs. Small question, is it setup to also comment on the PRs with issues found, like I see on the keycloak-gatekeeper PRs e.g. [keycloak/keycloak-gatekeeper#473](https://github.com/keycloak/keycloak-gatekeeper/pull/473)
> 

Thank you and yes that's correct, golangci-bot will comment on PRs if there are any issues.

> Reading through the issue Peter linked, it looks like the git.apache.org infrastructure has had serious issues. I don't see any issues with adding `replace git.apache.org/thrift.git => github.com/apache/thrift v0.12.0` to the go.mod file now, it would be frustrating to have our CI fail randomly.

Agreed. I did the rebase, based on the latest changes from @pb82. Please let me know if you have more questions.
